src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,180 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = tail = new Node(null, value, null);
+        } else {
+            Node notTheTailAnyMore = tail;
+            tail = new Node(notTheTailAnyMore, value, null);
+            notTheTailAnyMore.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
```

Comment:
Common mistake: Don't complicate if-else construction. In this case, you should remove the `else` clause because it's redundant. If the condition in the `if` clause is true, the method will exit due to the `throw` statement.